protocol/msgpack/unaggregated_iterator.go

```diff
@@ -166,12 +173,28 @@ func (it *unaggregatedIterator) decodeBatchTimer() {
 	}
 	it.metric.Type = unaggregated.BatchTimerType
 	it.metric.ID = it.decodeID()
-	numValues := it.decodeArrayLen()
-	values := it.floatsPool.Get(numValues)
+	var (
+		timerValues []float64
+		poolAlloc   = false
+		numValues   = it.decodeArrayLen()
+	)
+	if cap(it.timerValues) >= numValues {
+		it.timerValues = it.timerValues[:0]
+		timerValues = it.timerValues
+	} else if numValues <= it.largeFloatsSize {
+		it.timerValues = make([]float64, 0, numValues)
```

Comment:
In case the smaller timer values are sparse enough (you'll have to emit 1K/s values given the current defaults), we may not reach `largeFloatsSize`. I'd rather let it take its course and get to the `largeFloatsSize` naturally, given that even reallocation happens, it shouldn't be too bad. I can however allocate a smaller size and let the new capacity be the bigger of `numValues` and `2 * cap(it.timerValues)`